website/server/models/group.js

```diff
@@ -317,11 +317,90 @@ schema.statics.getGroups = async function getGroups (options = {}) {
   return groupsArray;
 };
 
+async function translateSystemMessages (group, user) {
+  let usernames = {};
+  let foundText = '';
+  for (let i = 0; i < group.chat.length; i++) {
+    if (!_.isEmpty(group.chat[i].info)) {
+      if (_.has(group.chat[i].info, 'user') && !_.has(usernames, group.chat[i].info.user)) {
+        usernames[group.chat[i].info.user] = null;
+      } else if (_.has(group.chat[i].info, 'target') && !_.has(usernames, group.chat[i].info.target)) {
+        usernames[group.chat[i].info.target] = null;
+      }
+    }
+  }
+  await Bluebird.map(Object.keys(usernames), async (username) => {
```

Comment:
two things: can you remove Bluebird and switch to native Promises? We're going to move away from Bluebird in the coming weeks. And I'm a bit worried about performances here, can we cache the username inside the info object instead of fetching it every time the chat is loaded?

Reply:
So you think it is better to save the username instead of the uuid? I saved the uuid to update the username automatically if the user updates it, but if you think it does not worth the extra processing, I can change it to the username.
If I do this change, the Promise won't be needed.

Reply:
@Yutsuten It's actually better to display the profile name that the player was using at the time that the boss damage was done. Chat messages made by the player themself always show the name at the time the post was made - they aren't updated later if the player changes their name.